src/main/java/com/couchbase/lite/Database.java

```diff
@@ -1966,7 +2143,7 @@ public long getStartTime() {
      */
     @InterfaceAudience.Private
     public boolean isOpen() {
-        return open.get();
+        return (open.get() && !closing.get()) || opening.get();
```

Comment:
If opening is true, doesn't it mean it's still opening but not done opening yet?